charts/kubefed/charts/controllermanager/templates/webhook.yaml

```diff
@@ -17,22 +23,25 @@ webhooks:
     caBundle: {{ b64enc $ca.Cert | quote }}
   rules:
   - operations:
-    - "CREATE"
-    - "UPDATE"
+    - CREATE
+    - UPDATE
     apiGroups:
-    - "core.kubefed.k8s.io"
+    - core.kubefed.k8s.io
     apiVersions:
-    - "v1beta1"
+    - v1beta1
     resources:
-    - "federatedtypeconfigs"
-    - "federatedtypeconfigs/status"
+    - federatedtypeconfigs
+    - federatedtypeconfigs/status
   failurePolicy: Fail
----
-apiVersion: admissionregistration.k8s.io/v1beta1
-kind: ValidatingWebhookConfiguration
-metadata:
-  name: "kubefedclusters.core.kubefed.k8s.io"
-webhooks:
+{{- if and .Values.global.scope (eq .Values.global.scope "Namespaced") }}
+# For namespace scoped deployments: filter admission webhook requests for
+# resources whose namespace matches the default namespace label applied by helm
+# upon creating the namespace. User must set this label manually if namespace
+# is created prior to installing chart with helm.
+  namespaceSelector:
+    matchLabels:
+      name: {{ .Release.Namespace }}
```

Comment:
Is it possible to use a more specific label?  I'm not sure it will be obvious to a user that a label `name: <namespace name>` is going to be critical to the functioning of the KubeFed API.  Ideally it would be possible to use something more specific (e.g. `kubefed-namespace: <namespace name>`).

Reply:
That was my original goal, but unfortunately, it is not currently possible with helm (see https://github.com/helm/helm/issues/4178 for more details). However, this suggests I should add a TODO about this to clarify.